pysat/_instrument.py

```diff
@@ -1597,7 +1648,8 @@ def load(self, yr=None, doy=None, date=None, fname=None, fid=None,
 
         # apply default instrument routine, if data present
         if not self.empty:
-            self._default_rtn(self)
+            # Does not require self as input, as it is a partial func
+            self._default_rtn()
```

Comment:
```suggestion
            self._default_rtn(self)
```

Reply:
I jumped the gun again on that suggestion above. I'm getting an error either way. Using your line,
```
In [2]: import pysat                                                                                                                                                                                                        

In [3]: jro = pysat.Instrument('jro', 'isr', 'drifts')                                                                                                                                                                      

In [4]: %pdb                                                                                                                                                                                                                
Automatic pdb calling has been turned ON

In [5]: jro.load(2010, 18)                                                                                                                                                                                                  
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-5-cd0bba3da139> in <module>
----> 1 jro.load(2010, 18)

~/Code/pysat/pysat/_instrument.py in load(self, yr, doy, date, fname, fid, verifyPad)
   1650         if not self.empty:
   1651             # Does not require self as input, as it is a partial func
-> 1652             self._default_rtn()
   1653 
   1654         # clean data, if data is present and cleaning requested

~/Code/pysatMadrigal/pysatMadrigal/instruments/methods/madrigal.py in filter_data_single_date(inst)
    661 
    662     # only do this if loading by date!
--> 663     if inst._load_by_date and inst.pad is None:
    664         # identify times for the loaded date
    665         idx, = np.where((inst.index >= inst.date)

AttributeError: 'NoneType' object has no attribute '_load_by_date'
> 
```
For some reason `None` is passed in for inst. If I put the self in, as I suggested, then I get a multiple values for self error. Using pdb I went back to line 1652 and `self` is just fine there. Also using pdf, at 663 inst is indeed `None`.

Reply:
The `filter_data_single_date` method is getting attached as a partial function, somehow. In that assignment inst is given a default of None. A couple other functions that aren't explicitly assigned via functools.partial are coming up as partial as well. This includes both `download` and `list_files`, and those those are working. Not sure what is going on here.

```
In [5]: jro._default_rtn                                                                                                                                                                                                    
Out[5]: functools.partial(<function filter_data_single_date at 0x7fbdb0a14040>, inst=None)

In [6]: jro._load_rtn                                                                                                                                                                                                       
Out[6]: <function pysatMadrigal.instruments.jro_isr.load(fnames, tag=None, inst_id=None, file_type='hdf5')>

In [7]: jro._clean_rtn                                                                                                                                                                                                      
Out[7]: <function pysatMadrigal.instruments.jro_isr.clean(self)>

In [8]: jro._download_rtn                                                                                                                                                                                                   
Out[8]: functools.partial(<function download at 0x7fbdb0a14b80>, user=None, password=None, file_type='hdf5')

In [9]: jro._init_rtn                                                                                                                                                                                                       
Out[9]: <function pysatMadrigal.instruments.jro_isr.init(self)>

In [10]: jro._list_remote_files_rtn                                                                                                                                                                                         
Out[10]: functools.partial(<function list_remote_files at 0x7fbdb0a15f70>, supported_tags={'': {'drifts': 'jro{{year:4d}}{{month:02d}}{{day:02d}}drifts.{{version:03d}}.{file_type}', 'drifts_ave': 'jro{{year:4d}}{{month:02d}}{{day:02d}}drifts_avg.{{version:03d}}.{file_type}', 'oblique_stan': 'jro{{year:4d}}{{month:02d}}{{day:02d}}.{{version:03d}}.{file_type}', 'oblique_rand': 'jro{{year:4d}}{{month:02d}}{{day:02d}}?.{{version:03d}}.{file_type}', 'oblique_long': 'jro{{year:4d}}{{month:02d}}{{day:02d}}?.{{version:03d}}.{file_type}'}}, inst_code=10)

In [11]: jro._list_files_rtn                                                                                                                                                                                                
Out[11]: functools.partial(<function list_files at 0x7fbdb0a14820>, file_type='hdf5')

```

So I just noticed how I'm behind :) I was focused on tracking down a load problem from the other pull request and didn't see any of the earlier comments.